content/articles/creating-a-shopping-cart-api/index.md

```diff
@@ -0,0 +1,580 @@
+#### AdminDao
+```java
+@Repository
+public class AdminDao {
+    Map<Long, Admin> adminDao=new HashMap<>();
+    public Admin registerAdmin(Long id,Admin admin){
+      return   adminDao.put(id,admin);
+    }
+    public void removeAdmin(Long id){
+        adminDao.remove(id);
+    }
+    public Admin getAdmin(Long id){
+        return adminDao.get(id);
+    }
+    public Map<Long,Admin> getAllAdmin(){
+        return adminDao;
+    }
+}
+```
+#### CustomerDao
+```java
+@Repository
+public class CustomerDao {
+
+    Map<Long, Customer> customerDao=new HashMap<>();
+    public Customer registerCustomer(Long id,Customer customer){
+       return customerDao.put(id,customer);
+    }
+    public void removeCustomer(Long id){
+        customerDao.remove(id);
+    }
+    public Customer getCustomer(Long id){
+        return customerDao.get(id);
+    }
+
+public int totalNumberOfCustomers(){
+        return customerDao.size();
+}
+public Map<Long,Customer> gettingAllTheCustomers(){
+        return customerDao;
+}
+
+}
+```
+#### ProductDao
+```java
+@Repository
+public class ProductDao {
+    Map<Long, Product> productDao=new HashMap<>();
+    // The products are stored in the product dao like that of a shelf in the super market
+    public void customerCanAddProduct(Long id,Product product){
+        productDao.put(id,product);
+    }
+    public void customerCanRemoveProduct(Long id,Product product){
+        productDao.remove(id,product);
+    }
+    public Product customerCanGetProduct(Long id,Product product){
+        return productDao.get(id);
+    }
+}
+```
+N/B At the top of the class, you must put the `@Repository` annotation to let spring know that that particular class is a repository that maps to entities to the database.
+### Creating the service layer
+The service layer is where we create our interface and also implement it. This layer contains all our business logic and you must always but the `@Service` at either in the interface or the implementation. Now, we will have a look at our interfaces and later their implementations.
+The service layer contains the following interfaces:
+#### CartService
+```java
+
+public interface CartService {
+    void addProductToCartByCustomer(Long id, Product product);
+    void addProductToCartByAdmin(Long id, Product product);
+    void removeProductsFromCartByCustomer(Long id, Product product);
+    void removeProductsFromCartByAdmin(Long id, Product product);
+    List<Product> checkAllTheProductsInTheCartByCustomer(Long id);
+    List<Product> checkAllTheProductsInTheCartByAdmin(Long id);
+    BigDecimal calculatePrice();
+    int totalNumberOfItems();
+    BigDecimal calculatingPurchase(Long id);
+}
+}
```

Comment:
This code doesn't compile because of the extra curly brace (i.e. line 187). Also, reformat your code for this class.